### PR TITLE
Refactor/extensible questions

### PR DESCRIPTION
**Pull request by Si3rr4wow**
## Ticket Number

https://pins-ds.atlassian.net/browse/AAPD-

## Checklist

- [ ] Requires infrastructure changes
- [ ] I have updated the documentation accordingly
- [ ] My commit history in this PR is linear
- [ ] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `main` (please only [rebase](https://github.com/Planning-Inspectorate/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: packages/forms-web-app/src/dynamic-forms/dynamic-components/list-add-more/question.js [hunk 1]
```diff
@@ -321,6 +321,32 @@ class ListAddMoreQuestion extends Question {
 	async removeAction(req, journeyResponse, answerId) {
 		return this.subQuestion.removeList(req, journeyResponse, answerId);
 	}
+
+	/**
+	 * @param {JourneyResponse} journeyResponse
+	 * @returns {boolean}
+	 */
+	isAnswered(journeyResponse) {
```
Comment (Si3rr4wow) [inline packages/forms-web-app/src/dynamic-forms/dynamic-components/list-add-more/question.js:329]: You're probably wondering why the heck I did this here. Honestly it didn't need doing immediately but by creating an isAnswered method on questions I was able to cut out a lot of logic from `Section.getStatus` (which is where this code is ripped from) and remove Sections dependencies on add more questions and validators.

File: packages/forms-web-app/src/dynamic-forms/dynamic-components/multi-file-upload/question.js [hunk 2]
```diff
@@ -40,32 +44,30 @@ class MultiFileUploadQuestion extends Question {
 	 * @param {string} [params.url]
 	 * @param {string} [params.pageTitle]
 	 * @param {string} [params.description]
-	 * @param {Object} [params.documentType]
+	 * @param {import('@pins/common/src/document-types').DocType} [params.documentType]
 	 * @param {string} [params.html]
-	 * @param {Array.<import('../../question').BaseValidator>} [params.validators]
+	 * @param {Array<import('../../question').BaseValidator>} [params.validators]
+	 *
+	 * @param {Array<Function>} [methodOverrides]
```
Comment (Si3rr4wow) [inline packages/forms-web-app/src/dynamic-forms/dynamic-components/multi-file-upload/question.js:51]: this should be a `Record`

File: packages/forms-web-app/src/dynamic-forms/dynamic-components/multi-file-upload/question.js [hunk 3]
```diff
@@ -107,16 +106,12 @@ class MultiFileUploadQuestion extends Question {
 	 * @returns {Promise<void>}
 	 */
 	async saveAction(req, res, journey, section, journeyResponse) {
-		// check for validation errors
-		const errorViewModel = this.checkForValidationErrors(req, section, journey);
-		if (errorViewModel) {
-			return this.renderAction(res, errorViewModel);
-		}
```
Comment (Si3rr4wow) [inline packages/forms-web-app/src/dynamic-forms/dynamic-components/multi-file-upload/question.js:114]: This code boggled my mind. It looks like it should do something except that `this.checkForValidationErrors` (as you'll see in a few lines) just returns void.

File: packages/forms-web-app/src/dynamic-forms/dynamic-components/multi-file-upload/question.js [hunk 4]
```diff
@@ -147,6 +142,8 @@ class MultiFileUploadQuestion extends Question {
 		return this.handleNextQuestion(res, journey, section.segment, this.fieldName);
 	}
 
+	// Belongs to save
```
Comment (Si3rr4wow) [inline packages/forms-web-app/src/dynamic-forms/dynamic-components/multi-file-upload/question.js:145]: breadcrumbs for myself in a later PR

File: packages/forms-web-app/src/dynamic-forms/dynamic-components/multi-file-upload/question.js [hunk 5]
```diff
@@ -240,102 +230,9 @@ class MultiFileUploadQuestion extends Question {
...
 	}
 
-	/**
-	 * returns the data to send to the DB
-	 * side effect: modifies journeyResponse with the new answers
-	 * @param {import('express').Request} req
-	 * @param {JourneyResponse} journeyResponse - current journey response, modified with the new answers
-	 * @returns {Promise.<Object>}
-	 */
-	async getDataToSave(req, journeyResponse) {
```
Comment (Si3rr4wow) [inline packages/forms-web-app/src/dynamic-forms/dynamic-components/multi-file-upload/question.js:250]: This moved out to the journeys/question-overrides/multi-file-upload.js

**Review by bryn-pins — APPROVED** [review]

**Pull request merged**